[import/named] fails to track deep exports when re-exporting from a module in node_modules

Here's a reproduction: https://github.com/kentcdodds/eslint-export-named-issue

```javascript
// exporter.js
// NOTE: I observed that it's only a problem if you export * from a file in node_modules
// and it doesn't matter module it is. In my case, I'm trying to re-export from `@testing-library/react`.
// here I'll just use eslint-plugin-import so we don't need to install anything else.

import {config} from 'eslint-plugin-import' // <-- no error here (so it definitely exports `config`)

export * from 'eslint-plugin-import'

// importer.js

import {configs} from './exporter' // <-- import/named fails
```

I'm not sure why this is an issue, but like I said, it appears to only happen when you re-export from a `node_modules` module. Seems to work fine if you re-export from one of your own files.

We have the same problem, and I started poking around. Not entirely familiar with the code, but it seems to be the module is ignored due the unambiguous check. When importing the node_module module, it gets the compiled es5 source code, which fails the unambiguous test. So hasDeep has a null innerMap, and returns no exports, so the import is not found.

It seems possibly to be a similar condition to #1460.

We have gotten some modules do work with `export * from 'modulename'`, and I'm guessing they are resolving to the es6 module source instead of the es5.

Using Kent's repo, here is the log output:
```
% DEBUG=eslint-plugin-import* npm run lint    

> @ lint /Users/me/dev/sandbox/eslintexport2
> eslint .

  eslint-plugin-import:resolver:node Resolving: eslint-plugin-import from: /Users/me/dev/sandbox/eslintexport2/exporter.js +0ms
  eslint-plugin-import:resolver:node Resolved to: /Users/me/dev/sandbox/eslintexport2/node_modules/eslint-plugin-import/lib/index.js +3ms
  eslint-plugin-import:ExportMap ignored path due to unambiguous regex: /Users/me/dev/sandbox/eslintexport2/node_modules/eslint-plugin-import/lib/index.js +0ms
  eslint-plugin-import:resolver:node Resolving: ./exporter from: /Users/me/dev/sandbox/eslintexport2/importer.js +6ms
  eslint-plugin-import:resolver:node Resolved to: /Users/me/dev/sandbox/eslintexport2/exporter.js +1ms
  eslint-plugin-import:ExportMap cache miss e4390246ed48adc78ea6b4ca0d8aadeedb549dccd99cb9178c4f72a69f41013f for path /Users/me/dev/sandbox/eslintexport2/exporter.js +6ms

/Users/me/dev/sandbox/eslintexport2/importer.js
  1:9  error  configs not found in './exporter'  import/named

✖ 1 problem (1 error, 0 warnings)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ This change log adheres to standards from [Keep a CHANGELOG](http://keepachangel
 - [`extentions`]: Fix scope regex ([#1611], thanks [@yordis])
 - [`no-duplicates`]: allow duplicate imports if one is a namespace and the other not ([#1612], thanks [@sveyret])
 - Add some missing rule meta schemas and types ([#1620], thanks [@bmish])
+- [`named`]: for importing from a module which re-exports named exports from a `node_modules` module ([#1569], [#1447], thanks [@redbugz], [@kentcdodds])
 
 ### Changed
 - [`import/external-module-folders` setting] behavior is more strict now: it will only match complete path segments ([#1605], thanks [@skozin])
@@ -655,6 +656,7 @@ for info on changes for earlier releases.
 [#1589]: https://github.com/benmosher/eslint-plugin-import/issues/1589
 [#1586]: https://github.com/benmosher/eslint-plugin-import/pull/1586
 [#1572]: https://github.com/benmosher/eslint-plugin-import/pull/1572
+[#1569]: https://github.com/benmosher/eslint-plugin-import/pull/1569
 [#1563]: https://github.com/benmosher/eslint-plugin-import/pull/1563
 [#1560]: https://github.com/benmosher/eslint-plugin-import/pull/1560
 [#1551]: https://github.com/benmosher/eslint-plugin-import/pull/1551
@@ -670,6 +672,7 @@ for info on changes for earlier releases.
 [#1493]: https://github.com/benmosher/eslint-plugin-import/pull/1493
 [#1472]: https://github.com/benmosher/eslint-plugin-import/pull/1472
 [#1470]: https://github.com/benmosher/eslint-plugin-import/pull/1470
+[#1447]: https://github.com/benmosher/eslint-plugin-import/pull/1447
 [#1439]: https://github.com/benmosher/eslint-plugin-import/pull/1439
 [#1436]: https://github.com/benmosher/eslint-plugin-import/pull/1436
 [#1435]: https://github.com/benmosher/eslint-plugin-import/pull/1435
@@ -1089,3 +1092,5 @@ for info on changes for earlier releases.
 [@yordis]: https://github.com/yordis
 [@sveyret]: https://github.com/sveyret
 [@bmish]: https://github.com/bmish
+[@redbugz]: https://github.com/redbugz
+[@kentcdodds]: https://github.com/kentcdodds

--- a/src/ExportMap.js
+++ b/src/ExportMap.js
@@ -106,6 +106,7 @@ export default class ExportMap {
     if (name !== 'default') {
       for (let dep of this.dependencies) {
         let innerMap = dep()
+        if (innerMap == null) return { found: true, path: [this] }
         // todo: report as unresolved?
         if (!innerMap) continue
 

--- a/tests/files/re-export-common-star.js
+++ b/tests/files/re-export-common-star.js
@@ -1,0 +1,1 @@
+export * from './common'

--- a/tests/files/re-export-node_modules.js
+++ b/tests/files/re-export-node_modules.js
@@ -1,0 +1,1 @@
+export * from 'eslint'

--- a/tests/src/rules/named.js
+++ b/tests/src/rules/named.js
@@ -26,6 +26,7 @@ ruleTester.run('named', rule, {
     test({code: 'import { destructingRenamedAssign } from "./named-exports"'}),
     test({code: 'import { ActionTypes } from "./qc"'}),
     test({code: 'import {a, b, c, d} from "./re-export"'}),
+    test({code: 'import {RuleTester} from "./re-export-node_modules"'}),
 
     test({ code: 'import { jsxFoo } from "./jsx/AnotherComponent"'
          , settings: { 'import/resolve': { 'extensions': ['.js', '.jsx'] } } }),

--- a/tests/src/rules/named.js
+++ b/tests/src/rules/named.js
@@ -26,6 +26,7 @@ ruleTester.run('named', rule, {
     test({code: 'import { destructingRenamedAssign } from "./named-exports"'}),
     test({code: 'import { ActionTypes } from "./qc"'}),
     test({code: 'import {a, b, c, d} from "./re-export"'}),
+    test({code: 'import {a, b, c} from "./re-export-common-star"'}),
     test({code: 'import {RuleTester} from "./re-export-node_modules"'}),
 
     test({ code: 'import { jsxFoo } from "./jsx/AnotherComponent"'

